### PR TITLE
CORE-42 - Account Touched

### PR DESCRIPTION
**Pull request by atoff**
Closes CORE-42

### CODE REVIEW

**Comment by CalumTowers** [comment]
@atoff Thank you! I think this PR highlights the need for us to get a beta version of some our services (I think CTS and Forum at the very least) setup and configured on a staging area so that we can push branches to there if we want to test with some "proper" data vs Heroku.

**Comment by CalumTowers** [comment]
> @CalumTowers idk about codeclimate

Looks like we can make a simple change to stop the vast majority of its squawking. Could you rename the snake_case variables to be camelCase please? The rest of the issues it highlights are out of scope for this PR.

File: app/Jobs/Mship/SyncToCTS.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    private $account;
+
+    public function __construct(Account $account)
+    {
+        $this->account = $account;
+    }
+
+    public function handle()
+    {
+        $ctsDatabse = config('services.cts.database');
```
Comment (CalumTowers) [inline app/Jobs/Mship/SyncToCTS.php]: There is a small typo on `$ctsDatabase` in this class.

File: app/Listeners/Mship/SyncSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            // Prevent unnecessary executions
+            return;
+        }
+
+        \App\Jobs\Mship\SyncToCTS::dispatch($event->account);
+        \App\Jobs\Mship\SyncToHelpdesk::dispatch($event->account);
+        \App\Jobs\Mship\SyncToMoodle::dispatch($event->account);
+        \App\Jobs\Mship\SyncToForums::dispatch($event->account);
+
+        Log::debug($event->account->real_name.' was queued to sync to external services');
```
Comment (CalumTowers) [inline app/Listeners/Mship/SyncSubscriber.php]: It would be good to log the user's CID at this point too

File: app/Jobs/Mship/SyncToForums.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    private $account;
+
+    public function __construct(Account $account)
+    {
+        $this->account = $account;
+    }
+
+    public function handle()
```
Comment (CalumTowers) [inline app/Jobs/Mship/SyncToForums.php]: Should we consider putting a concern on the `Account` model that the account `HasForumAccount`?

File: app/Jobs/Mship/SyncToCTS.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    private $account;
+
+    public function __construct(Account $account)
+    {
+        $this->account = $account;
+    }
+
+    public function handle()
```
Comment (CalumTowers) [inline app/Jobs/Mship/SyncToCTS.php]: Should we consider putting a concern on the `Account` model that the account `HasCtsAccount`?

File: app/Listeners/Mship/SyncSubscriber.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * Register the listeners for the subscriber.
+     *
+     * @param  \Illuminate\Events\Dispatcher $events
+     */
+    public function subscribe($events)
+    {
+        $events->listen(
+                \App\Events\Mship\AccountAltered::class,
+                '\App\Listeners\Mship\SyncSubscriber@syncToAllServices'
```
Comment (CalumTowers) [inline app/Listeners/Mship/SyncSubscriber.php]: Do we need to define this or can we just register the listener within the service provider to fire on the `AccountAltered` event?

Comment (atoff) [reply app/Listeners/Mship/SyncSubscriber.php]: This method is better for the future if we want to only sync certain services for certain events, which I could imagine happening at some point?

**Comment by atoff** [comment]
> As a side note, should we consider deprecating the sync commands?

I wouldn't want to remove these entirely; the main reason I kept them was in case when this goes live and doesn't work we have a fall back. If your testing with the systems indicates its fine, I will update the command to fire a job instead?

**Pull request merged**